### PR TITLE
Regional registry endpoints

### PR DESCRIPTION
**Pull request by roleyfoley**
## Description
A collection of changes to introduce support for multi region registries through S3 based replication. This introduces a new deployment unit in the account level deployments `s3replica`. This unit should be deployed to secondary regions that you need registry artefacts available in. The existing `s3` deployment unit acts as the primary registry and is configured to sync the contents of it's bucket to all `s3replica` deployments in other regions. 

Also includes some minor fixes 
- Fix for output lookups when the resource type can not be found 
- Using the occurrence name for the build unit in URL registry pulls 

## Motivation and Context
The primary motivation for this is to allow for multi region deployments of components who require that their image artefacts are in the same region as their deployment i.e. AWS lambda functions 

## How Has This Been Tested?
Tested locally 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves the structure or operation of the implementation)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Followup Actions
- [x] None

## Checklist:
- [ ] My change requires a change to the [documentation](https://github.com/hamlet-io/docs).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] None of the above.

### CODE REVIEW

File: engine/commonApplication.ftl [hunk 1]
```diff
@@ -1,17 +1,35 @@
 [#ftl]
 
 [#-- Registry definitions --]
-[#function getRegistryEndPoint type occurrence ]
+
+[#function getRegistryEndpointValue occurrence qualfiers ]
```
Comment (ml019) [inline engine/commonApplication.ftl:5]:
```suggestion
[#function getRegistryEndpointValue occurrence qualifiers ]
```

File: engine/commonApplication.ftl [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
+
     [#return
         contentIfContent(
             getOccurrenceSettingValue(
-                occurrence, ["Registries", type, "Endpoint"], true),
+                occurrence, asFlattenedArray(["Registries", qualfiers, "Endpoint"]), true),
```
Comment (ml019) [inline engine/commonApplication.ftl:10]:
```suggestion
                occurrence, asFlattenedArray(["Registries", qualifiers, "Endpoint"]), true),
```

File: engine/commonApplication.ftl [hunk 3]
```diff
@@ -1,17 +1,35 @@
...
+            getOccurrenceSettingValue(
+                    occurrence, asFlattenedArray(["Registries", qualfiers, "Registry"]), true)
```
Comment (ml019) [inline engine/commonApplication.ftl:12]:
```suggestion
                    occurrence, asFlattenedArray(["Registries", qualifiers, "Registry"]), true)
```

File: legacy/account/account_cmk.ftl [hunk 4]
```diff
@@ -61,14 +61,6 @@
             [#assign volumeEncryptionKmsKeyName = formatName("account", "cmk", "volume", "encrypt")]
             [#assign volumeEncryptionKmsKeyAliasId = formatDependentResourceId(AWS_CMK_ALIAS_RESOURCE_TYPE, volumeEncryptionKmsKeyId)]
 
-            [#-- Check that service linked role exists --]
```
Comment (ml019) [inline legacy/account/account_cmk.ftl:64]: IS this part of some other change? Doesn't appear related to s3 replication?

Comment (roleyfoley) [reply legacy/account/account_cmk.ftl:64]: it was discovered as part of testing the deployment, I needed the CMK deployment to work in us-east-1 in order to setup the s3 replica bucket

**Review by ml019 — APPROVED** [review]
